app/src/main/java/com/eis/phoneringer/RingCommands/RingtoneHandler.java

```diff
@@ -0,0 +1,103 @@
+package com.eis.phoneringer.RingCommands;
+
+import android.content.Context;
+import android.media.AudioAttributes;
+import android.media.Ringtone;
+import android.media.RingtoneManager;
+import android.net.Uri;
+import android.os.Build;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+/**
+ * Class that interfaces with the RingtoneManager class of the Android library
+ *
+ * @author Alessandra Tonin, Luca Crema, Alberto Ursino
+ */
+public class RingtoneHandler {
+
+    private static Ringtone defaultRingtone;
```

Comment:
You should specify that this Ringtone is saved in order to have always the same instance of the object and be able to do Ringtone.stop() even if you lose/can't pass the reference of the Ringtone to the "stopper"

Reply:
Even though this could be delegated to another class